src/main/java/org/broadinstitute/hellbender/utils/test/VariantContextTestUtils.java

```diff
@@ -22,14 +24,40 @@
 import org.testng.Assert;
 
 import java.util.*;
-import java.util.regex.Pattern;
 import java.util.stream.Collectors;
-import java.util.stream.Stream;
 
 public final class VariantContextTestUtils {
 
     private VariantContextTestUtils() {}
 
+    /**
+     * Reads an entire VCF into memory, returning both its VCFHeader and all VariantContext records in
+     * the vcf. Supports both local files and NIO-supported remote filesystems such as GCS.
+     *
+     * For unit/integration testing purposes only! Do not call this method from actual tools!
+     *
+     * @param vcfPath path or URI to a VCF, as a String
+     * @return A Pair with the VCFHeader as the first element, and a List of all VariantContexts from the VCF
+     *         as the second element
+     */
+    public static Pair<VCFHeader, List<VariantContext>> readEntireVCFIntoMemory(final String vcfPath) {
```

Comment:
Returning this pair seems ugly to me.  Maybe better to return a little struct that has a `getVariants` and `getHeader`?  I hate seeing Pairs all over the place because the syntax is so nasty in java.

Reply:
I think the ugliness is bearable, and outweighed by the convenience of getting the header as well. Introducing a new class used only by this method seems like a worse option to me.